ecs-init/engine/engine.go

```diff
@@ -186,6 +214,7 @@ func (e *Engine) StartSupervised() error {
 				log.Error("could not upgrade agent", err)
 			}
 		}
+		retryCount++
```

Comment:
i think this change adds 2 seconds delay for update container agent api (happy case).. can you skip adding retry count when e.upgradAgent() does not return error?

Reply:
good catch, will do 👍